Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -405,18 +401,19 @@ String CDMInstanceOpenCDM::sessionIdByInitData(const InitData& initData, bool fi
     GST_MEMDUMP("init data", reinterpret_cast<const uint8_t*>(initData.characters8()), initData.length());
 
     String result;
+
     for (const auto& pair : m_sessionsMap) {
-        const std::string& sessionId = pair.first;
-        const Session& session = pair.second;
-        if (session.containsInitData(initData)) {
-            result = String::fromUTF8(sessionId.c_str());
+        const String& sessionId = pair.key;
+        const RefPtr<Session>& session = pair.value;
+        if (session->containsInitData(initData)) {
+            result = sessionId;
             break;
         }
     }
 
     if (result.isEmpty()) {
         if (firstInLine) {
-            result = String::fromUTF8(m_sessionsMap.begin()->first.c_str());
+            result = m_sessionsMap.begin()->key;
```

Comment:
I'll bother you a bit more here. The idea of this is not just getting the first in line, but getting the first in line if there's only one element. We should WARN if, when `firstInLine`, there is more than one element because there would be the chance that we mix sessions.

`firstInLine` should have a better name like `shouldReturnOnlySessionAsFallback`.